test/user.test.js

```diff
@@ -1763,6 +1763,367 @@ describe('User', function() {
     });
   });
 
+  describe('Email Update', function() {
+    describe('User changing email property', function() {
+      var usersId;
+      var currentEmailCredentials = { email: 'm@c.com', password: 'bar' };
+      var updatedEmailCredentials = { email: 'n@d.com', password: 'bar' };
+
+      beforeEach(function(done) {
+        async.series([
+          function(next) {
+            User.create({ email: currentEmailCredentials.email,
+            password: currentEmailCredentials.password }, function(err, user) {
+              usersId = user.id;
+              next();
+            });
+          },
+          function(next) {
+            User.login({ email: currentEmailCredentials.email,
+            password: currentEmailCredentials.password }, function(err, accessToken1) {
+              if (err) return next(err);
+              assert(accessToken1.userId);
+              next();
+            });
+          },
+          function(next) {
+            User.login({ email: currentEmailCredentials.email,
+            password: currentEmailCredentials.password }, function(err, accessToken2) {
+              if (err) return next(err);
+              assert(accessToken2.userId);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+
+      afterEach(function(done) {
+        User.deleteById(usersId, function(err, user) {
+          if (err) return done (err);
+          done();
+        });
+      });
+
+      it('invalidates sessions once the email is changed using `updateAttribute`', function(done) {
+        async.series([
+          function(next) {
+            User.findById(usersId, function(err, userFound)  {
+              if (err) return next(err);
+              userFound.updateAttribute('email', updatedEmailCredentials.email,
+            function(err, userInstance) {
+              if (err) return next(err);
+              assert.equal(userInstance.email, updatedEmailCredentials.email);
```

Comment:
No need to check that `updateAttribute` does it job, that's already tested elsewhere. I hink you can pass `next` directly as `updateAttribute` callback.

Once you apply https://github.com/strongloop/loopback/pull/2693/files#r77507957, it should be possible to simplify the tests so that `async.series` is no longer necessary.

```js
it(..., function(done) {
  user.updateAttribute(..., function(err, updated) {
    if (err) return done(err);
    assertNoAccessTokens(done);
  });
});
```

I would personally use Promises instead and simplify the code even more.

```js
it(..., function() {
  return user.updateAttributes(...)
    .then(assertNoAccessTokens);
});
```